Remove `orderedset` from camkes-deps

orderedset can't be set up properly in python3.11, even though it is not referred to in the python tools. However, keeping it in camkes-deps https://github.com/seL4/camkes-tool/blob/7c3165356fdc378f71315ecb952a54dd48bfa56e/tools/python-deps/setup.py#L32

breaks the pip install.

```
Building wheel for orderedset (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [19 lines of output]
      running bdist_wheel
      The [wheel] section is deprecated. Use [bdist_wheel] instead.
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-311
      creating build/lib.linux-x86_64-cpython-311/orderedset
      copying lib/orderedset/__init__.py -> build/lib.linux-x86_64-cpython-311/orderedset
      running build_ext
      building 'orderedset._orderedset' extension
      creating build/temp.linux-x86_64-cpython-311
      creating build/temp.linux-x86_64-cpython-311/lib
      creating build/temp.linux-x86_64-cpython-311/lib/orderedset
      x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/python3.11 -c lib/orderedset/_orderedset.c -o build/temp.linux-x86_64-cpython-311/lib/orderedset/_orderedset.o
      lib/orderedset/_orderedset.c:209:12: fatal error: longintrepr.h: No such file or directory
        209 |   #include "longintrepr.h"
            |            ^~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
```

--- a/.github/workflows/push.yml
+++ b/.github/workflows/push.yml
@@ -25,6 +25,8 @@ jobs:
       - uses: seL4/ci-actions/link-check@master
         with:
           exclude: js/node_modules
+          # produces 403 for link checker:
+          exclude_urls: "http://jinja.pocoo.org/docs/"
 
   style:
     name: Style

--- a/tools/check_deps.py
+++ b/tools/check_deps.py
@@ -157,7 +157,7 @@ DEPENDENCIES = {
                       PythonModule('plyplus', 'Python parsing module'),
                       PythonModule('ply', 'Python parsing module'),
                       PythonModule('elftools', 'Python ELF parsing module'),
-                      PythonModule('ordered_set', 'Python OrderedSet module (orderedset)'),
+                      PythonModule('ordered_set', 'Python OrderedSet module (ordered-set)'),
                       PythonModuleWith('six', 'Python 2/3 compatibility layer', 'assertCountEqual'),
                       PythonModule('sqlite3', 'Python SQLite module'),
                       PythonModule('pyfdt', 'Python flattened device tree parser')),

--- a/tools/python-deps/setup.py
+++ b/tools/python-deps/setup.py
@@ -10,7 +10,7 @@ Setup script for dependency metapackage.
 
 To add a python dependency, add it to the DEPS list below.
 
-To publish using these instructions, you need the virtualenv package 
+To publish using these instructions, you need the virtualenv package
 installed, and a properly set up ~/.pypirc file.
 
 To publish to pypitest:
@@ -29,7 +29,6 @@ DEPS = [
     'aenum',
     'jinja2>=3.0.0',
     'ordered-set',
-    'orderedset',  # For older source trees: remove in 0.7.4
     'plyplus',
     'pyelftools',
     'sel4-deps',
@@ -43,7 +42,7 @@ DEPS = [
 
 setup(
     name='camkes-deps',
-    version='0.7.3',
+    version='0.7.4',
     description='Metapackage for downloading build dependencies for CAmkES',
     long_description="""
 The CAmkES tool has many python dependencies.  This package depends on them all
@@ -58,4 +57,5 @@ in directory https://github.com/seL4/camkes-tool/tree/master/tools/python-deps
     author='TrustworthySystems',
     author_email='pypi@trustworthy.systems',
     install_requires=DEPS,
+    python_requires='>=3'
 )

